packages/design-system-react/src/components/icon/README.mdx

```diff
@@ -0,0 +1,50 @@
+import { Controls, Canvas } from '@storybook/blocks';
+import * as IconStories from './Icon.stories';
+
+# Icon
+
+The Icon component is used to render SVG icons with consistent styling and behavior.
+
+```tsx
+import { Icon, IconName } from '@metamask/design-system-react';
+
+<Icon name={IconName.AddSquare} />;
+```
```

Comment:
Add this to code example at top of docs to component template https://github.com/MetaMask/metamask-design-system/pull/139